libs/config/path_string_test.go

```diff
@@ -0,0 +1,84 @@
+package config_test
+
+import (
+	"fmt"
+	"testing"
+
+	. "github.com/databricks/cli/libs/config"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNewPathFromString(t *testing.T) {
+	for _, tc := range []struct {
+		input  string
+		output Path
+		err    error
+	}{
+		{
+			input:  "",
+			output: NewPath(),
+		},
+		{
+			input:  ".",
+			output: NewPath(),
+		},
+		{
+			input:  "foo.bar",
+			output: NewPath(Key("foo"), Key("bar")),
+		},
+		{
+			input:  "[1]",
+			output: NewPath(Index(1)),
+		},
+		{
+			input:  "foo[1].bar",
+			output: NewPath(Key("foo"), Index(1), Key("bar")),
+		},
+		{
+			input:  "foo.bar[1]",
+			output: NewPath(Key("foo"), Key("bar"), Index(1)),
+		},
+		{
+			input:  "foo.bar[1][2]",
+			output: NewPath(Key("foo"), Key("bar"), Index(1), Index(2)),
+		},
+		{
+			input:  "foo[1234]",
+			output: NewPath(Key("foo"), Index(1234)),
+		},
+		{
+			input: "foo[123",
+			err:   fmt.Errorf("invalid path: foo[123"),
+		},
+		{
+			input: "foo[foo]",
+			err:   fmt.Errorf("invalid path: foo[foo]"),
+		},
+		{
+			input: "foo..bar",
+			err:   fmt.Errorf("invalid path: foo..bar"),
+		},
+		{
+			input: "foo.bar.",
+			err:   fmt.Errorf("invalid path: foo.bar."),
+		},
+		{
+			// Every component may have a leading dot.
+			input:  ".foo.[1].bar",
+			output: NewPath(Key("foo"), Index(1), Key("bar")),
+		},
+		{
+			// But after an index there must be a dot.
+			input: "foo[1]bar",
```

Comment:
Spaces are not allowed in the path, is it correct?

Reply:
This is not covered in the parsing code. They will become part of the keys where applicable (i.e. in the middle of a key, or after an index).

Do you think it's worth defining rules for these?

Reply:
I think we can follow up on this later including addressing dots in the key names (if we have a case for this)